src/hotspot/share/opto/macro.cpp

```diff
@@ -1242,8 +1242,7 @@ void PhaseMacroExpand::expand_allocate_common(
     initial_slow_test = BoolNode::make_predicate(initial_slow_test, &_igvn);
   }
 
-  if (C->env()->dtrace_alloc_probes() ||
```

Comment:
This check was introduced in JDK-6788527.
I'm not very clear about the root cause.

Reply:
@vnkozlov do you remember?

Reply:
Yes, it replaced `DTraceAllocProbes` flag check with cached value:
https://hg.openjdk.java.net/jdk7u/jdk7u-dev/hotspot/rev/c96bf21b756f

We cached JVMTI and DTrace flags values before compilation and check after compilation to throw it out if flags were changed:
https://hg.openjdk.java.net/jdk7u/jdk7u-dev/hotspot/rev/c96bf21b756f#l8.38